### PR TITLE
Fix bug in gitlab run images

### PR DESCRIPTION
**Pull request by jpbusch**
- fix default command for run images, build with gitlab ci
- fix handling of `${CI_RUNNER_EXECUTABLE_ARCH}` (default runner architecure) by checking the platform variable in job:script instead of job:rule
- improve entrypoint analogous to https://github.com/ika-rwth-aachen/docker-ros-ml-images/pull/1

### CODE REVIEW

File: .gitlab-ci/docker-ros.yml [hunk 1]
```diff
@@ -77,6 +77,7 @@ default:
     - privileged
```
Comment (lreiher) [inline .gitlab-ci/docker-ros.yml:77]: please add a PR description

File: .gitlab-ci/docker-ros.yml [hunk 2]
```diff
@@ -77,6 +77,7 @@ default:
...
     - amd64
   before_script:
+    - if [[ ${_PLATFORM} ]]  && ! [[ "${PLATFORM}" =~ ${_PLATFORM} ]]; then echo "job will not run for this architecture (${_PLATFORM})" && exit 0; fi
```
Comment (lreiher) [inline .gitlab-ci/docker-ros.yml:80]:
```suggestion
    - if [[ -n ${_PLATFORM} ]]  && ! [[ "${PLATFORM}" =~ ${_PLATFORM} ]]; then echo "Platform '${_PLATFORM}' is skipped; can be enabled by adding '${_PLATFORM}' to the 'PLATFORM' variable" && exit 0; fi
```

File: .gitlab-ci/docker-ros.yml [hunk 3]
```diff
@@ -100,16 +101,16 @@ default:
...
         docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
       fi
-    - TARGET=${_TARGET} PLATFORM=${_PLATFORM} ./docker/docker-ros/scripts/ci.sh
-    - echo -e "section_start:`date +%s`:push_section[collapsed=true]\r\e[0K[docker-ros] Push ${IMAGE}"
-    - docker push ${IMAGE}
-    - echo -e "section_end:`date +%s`:push_section\r\e[0K"
+      TARGET=${_TARGET} PLATFORM=${_PLATFORM} ./docker/docker-ros/scripts/ci.sh
+      echo -e "section_start:`date +%s`:push_section[collapsed=true]\r\e[0K[docker-ros] Push ${IMAGE}"
+      docker push ${IMAGE}
+      echo -e "section_end:`date +%s`:push_section\r\e[0K"
```
Comment (lreiher) [inline .gitlab-ci/docker-ros.yml:107]: is this change needed?

**Review by lreiher — CHANGES_REQUESTED** [review]

File: .gitlab-ci/docker-ros.yml (hunk 3 above)
Comment (jpbusch) [reply .gitlab-ci/docker-ros.yml:107]: no, but it prettifies the log of th job

File: .gitlab-ci/docker-ros.yml [hunk 4]
```diff
@@ -172,6 +171,7 @@ run-arm64:
     AFTER_INIT_EMBED: git config --global url.https://${GIT_HTTPS_USER}:${GIT_HTTPS_PASSWORD}@${GIT_HTTPS_SERVER}.insteadOf https://${GIT_HTTPS_SERVER}
     DOCKER_RUN_OPTS: -u root:root
   before_script:
+    - if [[ ${_PLATFORM} ]]  && ! [[ "${PLATFORM}" =~ ${_PLATFORM} ]]; then echo "job will not run for this architecture (${_PLATFORM})" && exit 0; fi
```
Comment (lreiher) [inline .gitlab-ci/docker-ros.yml:174]:
```suggestion
    - if [[ -n ${_PLATFORM} ]]  && ! [[ "${PLATFORM}" =~ ${_PLATFORM} ]]; then echo "job will not run for this architecture (${_PLATFORM})" && exit 0; fi
```
